File: src/main/java/io/github/tt432/machinemax/common/entity/MMEntities.java
package io.github.tt432.machinemax.common.entity;

import io.github.tt432.machinemax.MachineMax;
import io.github.tt432.machinemax.common.entity.old.entity.AE86Entity;
import io.github.tt432.machinemax.common.entity.old.entity.TestCarEntity;
import io.github.tt432.machinemax.common.entity.part.MMPartEntity;
import net.minecraft.core.registries.Registries;
import net.minecraft.world.entity.EntityType;
import net.minecraft.world.entity.MobCategory;
import net.neoforged.neoforge.registries.DeferredRegister;

import java.util.function.Supplier;

/**
 * 此类为实体注册器
 * 收录了模组添加的所有实体，并定义一些基本属性
 * @author 甜粽子
 */
public class MMEntities {
    public static void register() {}
    //以下为注册的实体列表
    //测试用车辆
    public static final Supplier<EntityType<TestCarEntity>> TEST_CAR_ENTITY =
            MachineMax.REGISTER.<TestCarEntity>entityType()
                    .id("test_car")
                    .builder(EntityType.Builder.of(TestCarEntity::new, MobCategory.MISC)
                    .fireImmune())
                    .build();
    //测试用越野车辆
    public static final Supplier<EntityType<AE86Entity>> AE86_ENTITY =
            MachineMax.REGISTER.<AE86Entity>entityType()
                    .id("ae86")
                    .builder(EntityType.Builder.of(AE86Entity::new, MobCategory.MISC)
                    .fireImmune())
                    .build();
    //测试用实体方块
    public static final Supplier<EntityType<MMPartEntity>> MM_PART_ENTITY =
            MachineMax.REGISTER.<MMPartEntity>entityType()
                    .id("part")
                    .builder(EntityType.Builder.<MMPartEntity>of(MMPartEntity::new, MobCategory.MISC)
                    .fireImmune()
                    .noSummon())
                    .build();
    //以上为注册的实体列表
}

File: src/main/kotlin/io/github/tt432/machinemax/common/registry/MMEntities.kt
package io.github.tt432.machinemax.common.registry

import io.github.tt432.machinemax.MachineMax
import io.github.tt432.machinemax.common.entity.MMPartEntity
import net.minecraft.world.entity.EntityType
import net.minecraft.world.entity.MobCategory

object MMEntities {
    @JvmStatic
    fun register() {
    }

    //部件实体
    @JvmStatic
    val PART_ENTITY = MachineMax.REGISTER.entityType<MMPartEntity>().id("part")
        .builder(EntityType.Builder.of(::MMPartEntity, MobCategory.MISC).fireImmune().noSummon().noSave().eyeHeight(0.0F))
        .build()

//    //载具核心实体
//    @JvmStatic
//    val CORE_ENTITY = MachineMax.REGISTER.entityType<CoreEntity>().id("core")
//        .builder(EntityType.Builder.of(::CoreEntity, MobCategory.MISC).fireImmune().noSummon())
//        .build()
}